oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -134,11 +134,11 @@ public List<OSProcess> getDescendantProcesses(int parentPid, Predicate<OSProcess
         // Get this pid and its descendants
         List<OSProcess> descendantProcs = queryDescendantProcesses(parentPid);
         // Extract the parent from the list
-        OSProcess parent = descendantProcs.stream().filter(p -> p.getProcessID() == parentPid).findAny().orElse(null);
+        OSProcess parent = descendantProcs.stream().parallel().filter(p -> p.getProcessID() == parentPid).findAny().orElse(null);
```

Comment:
Similar to child process answer above. Just parallelizing a getter of a relatively small number, and finding any... single thread likely to stop more quickly when it finds one than trying to coordinate multiple threads which may each run until they find one and then have to talk to each other.....